graphql/src/query.rs

```diff
@@ -474,31 +475,47 @@ impl Mutation {
     }
 
     /// Adds a bookmark to the user's list of bookmarks.
-    async fn add_bookmark(&self, context: &Context<'_>, document_id: Uuid) -> FieldResult<String> {
+    #[graphql(guard = "AuthGuard")]
+    async fn add_bookmark(
+        &self,
+        context: &Context<'_>,
+        document_id: Uuid,
+    ) -> FieldResult<Option<AnnotatedDoc>> {
         let user = context
             .data_opt::<UserInfo>()
             .ok_or_else(|| anyhow::format_err!("User is not signed in"))?;
-        Ok(context
+        let document = context
+            .data::<DataLoader<Database>>()?
+            .load_one(dailp::DocumentId(document_id))
+            .await?;
+        context
             .data::<DataLoader<Database>>()?
             .loader()
             .add_bookmark(document_id, user.id)
-            .await?)
+            .await?;
+        Ok(document)
     }
 
     /// Removes a bookmark from a user's list of bookmarks
+    #[graphql(guard = "AuthGuard")]
     async fn remove_bookmark(
         &self,
         context: &Context<'_>,
         document_id: Uuid,
-    ) -> FieldResult<String> {
+    ) -> FieldResult<Option<AnnotatedDoc>> {
         let user = context
             .data_opt::<UserInfo>()
             .ok_or_else(|| anyhow::format_err!("User is not signed in"))?;
-        Ok(context
+        let document = context
+            .data::<DataLoader<Database>>()?
+            .load_one(dailp::DocumentId(document_id))
+            .await?;
+        context
             .data::<DataLoader<Database>>()?
             .loader()
             .remove_bookmark(document_id, user.id)
-            .await?)
+            .await?;
+        Ok(document)
```

Comment:
Same thing re: order of doing the mutation (actually changing data) and fetching from the database.